internal/rbd/rbd_attach.go

```diff
@@ -388,6 +388,33 @@ func appendKRbdDeviceTypeAndOptions(cmdArgs []string, userOptions string) []stri
 	return cmdArgs
 }
 
+// appendCrushLocationMapOptions converts crushLocationMap to the format
+// "read_from_replica=localize,crush_location=key1:value1|key2:value2"
+// and appends it to mapOptions.
+func appendCrushLocationMapOptions(mapOptions string, crushLocationMap map[string]string) string {
```

Comment:
Why generate this string on every mount? It can be generated once at startup, because the crushLocationMap never changes while the node-plugin is running. It would be much nicer if the NodeServer has a function to return the map, or maybe even the generated string.